### PR TITLE
more explanation of sql field, plus changes to reflect how it changed in 5.2

### PR DESCRIPTION
**Pull request by robbiejackson**
How the SQL field worked was changed in 5.2, so these changes reflect that.

Also there's more explanation of how the linked fields mechanism works, and how the sample component makes use of that feature.

### CODE REVIEW

File: versioned_docs/version-5.2/general-concepts/forms-fields/standard-fields/sql.md [hunk 1]
```diff
@@ -427,8 +419,34 @@ To **redisplay the form** you can either send the form as an HTTP response to th
...
+
+### Example Component
+
+You can download [this com_sqlfield component](./_assets/com_sqlfield.zip) as an example to follow.
+
+Go onto your Joomla instance back-end, display the article categories and find the id of your `Uncategorised` category.
+
+If it's not 2 then edit the com_sqlfield site/forms/example_form.xml and set the `sql_default_catid` to whatever id it is.
+
+Install the com_sqlfield component and run the form by navigating to your Joomla instance's site page index.php/component/sqlfield/
```
Comment (HLeithner) [inline versioned_docs/version-5.2/general-concepts/forms-fields/standard-fields/sql.md:432]:
```suggestion
Install the com_sqlfield component and run the form by navigating to your Joomla instance's site page `index.php?option=sqlfield`
```

File: versioned_docs/version-5.2/general-concepts/forms-fields/standard-fields/sql.md [hunk 2]
```diff
@@ -427,8 +419,34 @@ To **redisplay the form** you can either send the form as an HTTP response to th
...
+
+The component will display an HTML select field with the article categories available on your instance, and in the next field, 
+the titles of the articles associated with the selected category.
+
+If you select a different category then the component will send the form data to the server in an HTTP POST request, 
+with the task parameter set to "sqlfield.reload" which will cause the `SqlfieldController::reload()` function to be called.
+This function stores the existing form values in the user state, and arranges for the form to be redisplayed. 
+
+When the component redisplays the form it injects the existing form values into the Joomla `Form` structure. 
+(This is standard practice, so that the when the form is redisplayed the user sees the values he/she previously entered).
```
Comment (HLeithner) [inline versioned_docs/version-5.2/general-concepts/forms-fields/standard-fields/sql.md:442]: > so that when the form

doesn't sound right

**Comment by HLeithner** [comment]
additional annotations:  the com_sqlfield should be moved to the manual-example repo and restructured and updated to the latest joomla (psr12) coding style if you have time.

File: versioned_docs/version-5.2/general-concepts/forms-fields/standard-fields/sql.md (hunk 1 above)
Comment (robbiejackson) [reply versioned_docs/version-5.2/general-concepts/forms-fields/standard-fields/sql.md:432]: Yes, I'll fix that where it occurs (needs to be option=com_sqlfield, of course), and add about using the other only if you have SEF URLs on.

File: versioned_docs/version-5.2/general-concepts/forms-fields/standard-fields/sql.md (hunk 2 above)
Comment (robbiejackson) [reply versioned_docs/version-5.2/general-concepts/forms-fields/standard-fields/sql.md:442]: Yes, it's a typo

**Comment by robbiejackson** [comment]
I wasn't sure about adding com_sqlfield to the manual examples repo because it's very like the com_exampleform component which is already there. I was worried that the repo would get too cluttered with similar extensions, and it might be better just to have a few well-documented examples there, but which didn't have a lot of overlap with others. But I'm rather undecided about it.

What do you think?

**Comment by HLeithner** [comment]
> I wasn't sure about adding com_sqlfield to the manual examples repo because it's very like the com_exampleform component which is already there. I was worried that the repo would get too cluttered with similar extensions, and it might be better just to have a few well-documented examples there, but which didn't have a lot of overlap with others. But I'm rather undecided about it.
> 
> What do you think?

My main reason behind this, not storing it not here, so doesn't get "forgotten", second reason is that it can be automatically tested with new joomla versions with our ci which also helps the cms not to break things unintentionally.

File: versioned_docs/version-5.2/general-concepts/forms-fields/standard-fields/sql.md (hunk 1 above)
Comment (HLeithner) [reply versioned_docs/version-5.2/general-concepts/forms-fields/standard-fields/sql.md:432]: don't make it complicated, I think here it's only relevant that you open the component so you don't need to care about sef, in my opinion. The documentation should be straightforward when it's possible without to much "sidenotes" the developer doesn't about in this moment.

we should try to stay with the KISS principals, at least my opinion.

**Pull request merged**